### ISSUE TITLE
Add argument to disable filter, transform and shuffling

### ISSUE THREAD

**Issue by LynxPDA** [LibreTranslate/Locomotive#13]
I kindly ask you to add an option (command line argument) to disable the filtering, transformation, augmentation and shuffling functions.

**Expected Behavior:**
For example, I already have pre-filtered corpuses, with sentences in the desired sequence, and I would like to use the corpus as is. And use additional options for new datasets and training.

**Actual behavior:**
With a corpus of over 140 million pairs of sentences, I couldn’t wait to start training. (more than 40 minutes on Ryzen 7950X).

```
{}
Filtered 0 lines out of 243096
Augmented 0
New sentence count: 921580
{}
Filtered 0 lines out of 116798
Augmented 0
New sentence count: 1038378
{}
Filtered 0 lines out of 325708
Augmented 0
New sentence count: 1364086
{}
Filtered 0 lines out of 886954
Augmented 0
New sentence count: 2251040
{}
Filtered 0 lines out of 895030
Augmented 0
New sentence count: 3146070
{}
Filtered 0 lines out of 3802600
Augmented 0
New sentence count: 6948670
{}
Filtered 0 lines out of 17478530
Augmented 0
New sentence count: 24427200
{}
Filtered 0 lines out of 48976102
Augmented 0
New sentence count: 73403302
{}
Filtered 0 lines out of 73140378
Augmented 0
New sentence count: 146543680
Training size: 146538680
Validation size: 5000
Writing shuffled sets
```

**Comment by LynxPDA** [comment]
The estimated processing time (based on the rate of increase in the size of the src.txt.shuffled and tgt.txt.shuffled files) is over 2600 minutes!
And I lost the opportunity to add additional data to the corpus and continue training with the new corpus.

**Comment by LynxPDA** [comment]
And how can we now assign different weights to different data bodies? As far as I understand, this no longer works.

**Comment by pierotofy** [comment]
Thanks for the feedback. I've come up with a strategy to allow more granular control by using weights:

## Using Weights

It's possible to specify weights for each source, for example, it's possible to instruct the training to use less samples for certain datasets:

{
    "sources": [
        {"source": "file://D:\\path\\to\\mydataset-en_es", "weight": 1},
        {"source": "http://data.argosopentech.com/data-ccaligned-en_es.argosdata", "weight": 5},
    ]
}

In the example above, 1 sample will be taken from mydataset and 5 will will be taken from CCAligned.

Specifying weights disables filtering, transformations and augmentations. The datasets are used as-is. No merging or shuffling is performed either. A weight of 1 can be used to instruct Locomotive to not preprocess a source.

In short, assign a `weight` property to any dataset that you don't want Locomotive to touch.
